src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -236,11 +236,14 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
 
 		    G4double ekin = energy - mass;
 
-		    particleGun->SetParticleEnergy(ekin);
+		    //particleGun->SetParticleEnergy(ekin);
```

Comment:
This was my mistake. I was comparing with what the particle gun does, and got it wrong...

@eosullivan The `Set*()` methods aren't Geant4 things, but are defined in the `WCSimPrimaryGeneratorAction` class, and are essentially storage so that the `jhfNtuple` can be filled correctly in `WCSimRunAction`. Calls to these `Set*()` methods were not there for muline, so the `jhfNtuple` (and therefore output file) was just getting filled with defaults (e.g. 0,0,0 for the vertex position)

The particleGun lines are how the muline input file is turned into physics events. They should be uncommented.

Reply:
@eosullivan I have uncommented those lines. Sorry for the confusion